lib/Dialect/FIRRTL/FIRRTLUtils.cpp

```diff
@@ -0,0 +1,114 @@
+    // Strict connect does not allow uninferred widths.
+    // TODO: enable strict connect, same thing below.
+    if (true || dstType.hasUninferredWidth())
+      builder.create<ConnectOp>(dst, src);
+    else
+      builder.create<StrictConnectOp>(dst, src);
+    return;
+  }
+
+  if (auto dstBundle = dstType.dyn_cast<BundleType>()) {
+    // Connect all the bundle elements pairwise.
+    auto numElements = dstBundle.getNumElements();
+    // Check if we are trying to create an illegal connect - just create the
+    // connect and let the verifier catch it.
+    auto srcBundle = srcType.dyn_cast<BundleType>();
+    if (!srcBundle || numElements != srcBundle.getNumElements()) {
+      builder.create<ConnectOp>(dst, src);
+      return;
+    }
+    for (size_t i = 0; i < numElements; ++i) {
+      auto dstField = builder.create<SubfieldOp>(dst, i);
+      auto srcField = builder.create<SubfieldOp>(src, i);
+      if (dstBundle.getElement(i).isFlip)
+        std::swap(dstField, srcField);
+      emitConnect(builder, dstField, srcField);
+    }
+    return;
+  }
+
+  if (auto dstVector = dstType.dyn_cast<FVectorType>()) {
+    // Connect all the vector elements pairwise.
+    auto numElements = dstVector.getNumElements();
+    // Check if we are trying to create an illegal connect - just create the
+    // connect and let the verifier catch it.
+    auto srcVector = srcType.dyn_cast<FVectorType>();
+    if (!srcVector || numElements != srcVector.getNumElements()) {
+      builder.create<ConnectOp>(dst, src);
+      return;
+    }
+    for (size_t i = 0; i < numElements; ++i) {
+      auto dstField = builder.create<SubindexOp>(dst, i);
+      auto srcField = builder.create<SubindexOp>(src, i);
+      emitConnect(builder, dstField, srcField);
+    }
+    return;
+  }
+
+  // Handle ground types with possibly uninferred widths.
+  auto dstWidth = dstType.getBitWidthOrSentinel();
+  auto srcWidth = srcType.getBitWidthOrSentinel();
+  if (dstWidth < 0 || srcWidth < 0) {
+    // If one of these types has an uninferred width, we connect them with a
+    // regular connect operation.
+    builder.create<ConnectOp>(dst, src);
+    return;
+  }
+
+  // The source must be extended or truncated.
+  if (dstWidth < srcWidth) {
+    // firrtl.tail always returns uint even for sint operands.
+    IntType tmpType = dstType.cast<IntType>();
+    if (tmpType.isSigned())
+      tmpType = UIntType::get(dstType.getContext(), dstWidth);
+    src = builder.create<TailPrimOp>(tmpType, src, srcWidth - dstWidth);
+    // Insert the cast back to signed if needed.
+    if (tmpType != dstType)
+      src = builder.create<AsSIntPrimOp>(dstType, src);
+  } else if (srcWidth < dstWidth) {
+    // Need to extend arg.
+    src = builder.create<PadPrimOp>(src, dstWidth);
+  }
+
+  // Strict connect requires the types to be completely equal, including
+  // connecting uint<1> to abstract reset types.
+  // TODO: enable strict connect.
+  if (false && dstType == srcType)
```

Comment:
This one should be safe right now.  reset uint<1> case will fall through to the connect.